packages/commonwealth/shared/utils.ts

```diff
@@ -282,3 +282,34 @@ export function aggregatePermissions(
   // compute permissions
   return BigInt(0); //bandaid fix. always allow permissions due to removing functionality.
 }
+
+/**
+ * Convert Cosmos-style minimal denom amount to readable full-denom amount
+ * Example 7000000 uosmo -> 7 OSMO
+ * Example 8000000000000000000 aevmos -> 8 EVMOS
+ *
+ */
+export function minimalToNaturalDenom(
+  amount?: string | number,
+  decimals?: number
+): number {
+  if (!amount || !decimals) return 0;
+  const amt = typeof amount === 'string' ? Number(amount) : amount;
```

Comment:
Will this number conversion cause overflow / rounding errors at large token values?

Reply:
You're right, it's not good for big numbers. Will use keplr's utils and add tests

Reply:
Unit tests added. LMK if there are some other cases I should consider. @jnaviask